hazelcast/src/main/java/com/hazelcast/nio/NodeIOService.java

```diff
@@ -40,15 +41,37 @@
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
 
 public class NodeIOService implements IOService {
 
     private final Node node;
     private final NodeEngineImpl nodeEngine;
 
+    private final AtomicInteger operationPacketsReceived;
```

Comment:
I'm sure we can come up with a more scalable approach. You can get rid of contention on shared AtomicInteger (make it AtomicLong) by giving each IO thread its own volatile long field.